client/src/components/formHelpers/form-fields.tsx

```diff
@@ -105,6 +110,10 @@ function FormFields({ formFields, options }: FormFieldsProps): JSX.Element {
                     placeholder={placeholder}
                     required={required.includes(name)}
                     rows={4}
+                    {...(isDisabled && {
+                      onInvalid: (e: React.FormEvent<HTMLInputElement>) =>
+                        e.preventDefault()
+                    })}
```

Comment:
needed to use this, to mimic `disabled` behavior, because we are using `aria-disabled`.

Reply:
the `disabled` attribute doesn't trigger an invalid event in the `input`, when the button is clicked, while the `aria-disabled` does because it doesn't drop or interact with the event if `aria-disabled` is true.

a hint for this is when the new button is clicked, it has an outline to indicate focused, while the old one doesn't.


---

> If we want to ignore the click event, would adding this logic to the `ui-components`'s Button be sufficient?

Depend on the path we take to make it not need in the `/learn`.

If we revert the morphing of `disabled` to `aria-disabled`, and let the consumers assign `disabled`, `aria-disabled`, and `aria-pressed` however they see fit, then it will be sufficient to not trigger the click event, so the invalid event won't trigger.

And if we disable the click event, entirely in the library, depend on how it applied, we may be creating `disabled` again, and removing the ability to read the button on specific screen readers, or create weird behaviors and edge cases.

And if we check "is there is a form input next to the button?" would lead to annoying behavior and coupling of concerns.

To be honest, I don't want to create these logics, and prefer to revert the `disabled` and `aria-disabled`, if this will be an issue, because I don't know how the button is going to be used in the future, I prefer letting the consumer deal with it.